src/components/structures/FavouriteMessagesView.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2015, 2016 OpenMarket Ltd
+Copyright 2017 Vector Creations Ltd
+Copyright 2018, 2019 New Vector Ltd
+Copyright 2019 - 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { MatrixClient, MatrixEvent } from 'matrix-js-sdk/src/matrix';
+import React, { useContext, useRef } from 'react';
+
+import MatrixClientContext from '../../contexts/MatrixClientContext';
+import { _t } from '../../languageHandler';
+import FavouriteMessageTile from '../views/rooms/FavouriteMessageTile';
+import ScrollPanel from './ScrollPanel';
+
+function getFavouriteMessagesTiles(cli: MatrixClient, favouriteMessageEvents) {
+    const ret = [];
+    let lastRoomId: string;
+
+    for (let i = (favouriteMessageEvents?.length || 0) - 1; i >= 0; i--) {
+        const timeline = [] as MatrixEvent[];
+        const resultObj = favouriteMessageEvents[i];
+        const roomId = resultObj.roomId;
+        const room = cli.getRoom(roomId);
+        const mxEvent = room.findEventById(resultObj.eventId);
+        timeline.push(mxEvent);
+
+        if (roomId !== lastRoomId) {
+            ret.push(<li key={mxEvent?.getId() + "-room"}>
+                <h2>{ _t("Room") }: { room.name }</h2>
+            </li>);
+            lastRoomId = roomId;
+        }
+
+        const resultLink = "#/room/"+roomId+"/"+mxEvent.getId();
+
+        ret.push(
+            <FavouriteMessageTile
+                key={mxEvent.getId()}
+                result={mxEvent}
+                resultLink={resultLink}
+                timeline={timeline}
+            />,
+        );
+    }
+    return ret;
+}
+
+let favouriteMessagesPanel;
+
+const FavouriteMessagesView = () => {
+    const favouriteMessageEvents= JSON.parse(
```

Comment:
We'll need to sort this by room somehow

Reply:
i am not sure, i get you properly here. Do you mean we will need a sorting functionality or what? cos if thats the case i am already planning to include a sorting functionality to the view after the second problem we discussed is resolved but its not going to be sorted by rooms but rather by the order the item is added to the list.

Reply:
Ah, I see, don't worry about it then